### DISCUSSION TITLE
Restore the fact that magit-push-current put origin/current-branch-name as first candidate

### DISCUSSION THREAD

**Discussion by damienrg** [magit/magit#5196] [Ideas]
Before a5ca96edcd48585c9ea4b97fbe332d5cb83dc34a, invoking `magit-push-current` would put `origin/current-branch-name` at the top of the list of the candidates. This was particularly useful with `helm` because it automatically selected `origin/current-branch-name` because it was the first one in the list.

The problem is due to the fact that `cl-union` is not stable:

```lisp
(cl-union '("foo") '("bar" "baz") :test #'equal) ;; ("foo" "bar" "baz")
(cl-union '("foo") '("bar" "baz" "foo") :test #'equal) ;; ("bar" "baz" "foo")
```
Reverting a5ca96edcd48585c9ea4b97fbe332d5cb83dc34a, fix the problem.

**Comment by tarsius** [comment]
This just happened to work by accident. It's also not the case that `origin/current` would always be returned. If a remote `abc` existed, for example, then `abc/current` would have been returned.

I do agree that `magit-push-current` should provide a better default, but simply reverting that change is not the solution. Something more sophisticated is required but that means this will have to wait in line with all the other feature requests.
